src/Amazonka/S3/StreamingUpload.hs

```diff
@@ -283,34 +284,43 @@ data S = S !Builder {-# UNPACK #-} !Int
 newS :: S
 newS = S mempty 0
 
-newSFrom :: ByteString -> S
-newSFrom bs = S (byteStringCopy bs) (B.length bs)
-
 appendS :: S -> ByteString -> S
 appendS (S builder len) bs = S (builder <> byteStringCopy bs) (len + B.length bs)
 
 finaliseS :: S -> IO ByteString
 finaliseS (S builder builderLen) = do
   fptr <- mallocForeignPtrBytes builderLen
   let ptr = unsafeForeignPtrToPtr fptr
-      bufWriter = runBuilder builder
-  bufWriter ptr builderLen >>= \case
+  runBuilder builder ptr builderLen >>= \case
     (written, Done)
       | written == builderLen -> pure $! PS fptr 0 builderLen
       | otherwise ->
           error $ "finaliseS: bytes written didn't match, expected: " <> show builderLen <> " got: " <> show written
```

Comment:
I imagine this has never fired in recent memory, but it seems more polite to throw a named exception instead of crashing the caller's program.

Reply:
This shouldn't be possible, but I left it there in case I screwed up really badly somehow... I should do something better though, an exception might be the way to go.